env/env_posix.cc

```diff
@@ -729,6 +771,62 @@ class PosixEnv : public Env {
     return result;
   }
 
+/**
+ * Loads the named library into the result.
+ * If the input name is empty, the current executable is loaded
+ * On *nix systems, a "lib" prefix is added to the name if one is not supplied
+ * Comparably, the appropriate shared library extension is added to the name if not supplied.
+ * If search_path is not specified, the shared library will be loaded using the default path (LD_LIBRARY_PATH)
+ * If search_path is specified, the shared library will be searched for in the directories
+ * provided by the search path
+ */
+Status LoadLibrary(const std::string& name,
+		     const std::string & path,
+		     std::shared_ptr<DynamicLibrary>* result) override {
+    Status status;
+    assert(result != nullptr);
+    if (name.empty()) {
+      void *hndl = dlopen(NULL, RTLD_NOW);
+      if (hndl != nullptr) {
+	result->reset(new PosixDynamicLibrary(name, hndl));
+	return Status::OK();
+      }
+    } else {
+      std::string library_name = name;
```

Comment:
hmm, I feel that we need to handle more cases when we try to determine the actual name of the shared object file. On both my machine and Travis machine, there is no 'libdl.so' in the designated path (not necessarily /usr/lib64), even though both are Linux. On my machine, there is 'libdl.so.2'. In many cases, I expect that libdl.so is a hard link to libdl.so.2, but I feel this assumption is strong, especially when the user has custom scripts.

Reply:
This test case was meant to look for a library on some standard path.  Since RocksDB does not ship with any shared libraries, there is no known library I can easily look for.  So I picked a system library (libdl.so) I knew would be present. I tried to use the test executable itself as the object to load, but that does not work on all platforms (MacOS works, RedHat does not).

The problem is that that library moves around from place to place on various systems.  There is no standard place for any of the system libraries.

I  updated that specific sub-test case to only look for a library on the path if we are building RocksDB as a shared library (ROCKSDB_DLL is set) and to look for the rocksdb shared library.   That will allow the test to pass in the case where we do not have a known shared library to look for.

Reply:
Not only does the library move, but its name can vary due to versioning as well. I don't think we can guarantee that there is a standard name, a.k.a, libdl.so for the library. Symlinks can be used, but I wonder if there is a better solution that does not require creation of symlink in a directory owned by root user.

Reply:
I guess my concern about `EnvPosixTest.LoadLibrary` is its assumption. This test assumes that we can find a shared object file whose name is exactly `libdl.so` in one of the directories searched by `dlopen` according to the manual. This assumption may not always be valid.

Reply:
I changed the test to use the "rocksdb" library.  If the library is not present (which it will not be unless a shared_lib build was done), the test will still pass.

Reply:
Thanks!